File: python/08_013.py
def first_uniq_char(s):

    # Count the occurrences of each character
    char_count = {}
    for char in s:
        char_count[char] = char_count.get(char, 0) + 1
    
    # Find the first character with count 1
    for i, char in enumerate(s):
        if char_count[char] == 1:
            return i
    
    return -1  # No unique character found

# Example usage
print(first_uniq_char("apple"))      # 0
print(first_uniq_char("loveapples"))  # 1
print(first_uniq_char("aabb"))        # -1

File: data structures and algorithms/cs/08_013.cs
using System;
using System.Collections.Generic;

public class FirstNonRepeatingChar {
    public static int FirstUniqChar(string s) {
        Dictionary<char, int> charCount = new Dictionary<char, int>();
        
        // Count the occurrences of each character
        foreach (char c in s) {
            if (charCount.ContainsKey(c)) {
                charCount[c]++;
            } else {
                charCount[c] = 1;
            }
        }
        
        // Find the first character with count 1
        for (int i = 0; i < s.Length; i++) {
            if (charCount[s[i]] == 1) {
                return i;
            }
        }
        
        return -1;  // No unique character found
    }
    
    public static void Main(string[] args) {
        Console.WriteLine(FirstUniqChar("apple"));      // 0
        Console.WriteLine(FirstUniqChar("loveapples")); // 1
        Console.WriteLine(FirstUniqChar("aabb"));       // -1
    }
}